### PR TITLE
2017 icd10 external data load files

### PR DESCRIPTION
**Pull request by stephenwaite**
hi Brady, here's the 10-1-16 files, i don't think they're using the reimbursement mappings anymore but if they release them at a later point i will be glad to add them.  I also changed the load to ignore the addenda files so they wouldn't interfere with the active value.

thanks

### CODE REVIEW

File: sql/4_2_2-to-4_3_1_upgrade.sql [hunk 1]
```diff
@@ -739,3 +739,18 @@ UPDATE `globals` SET `gl_value` = 'style_tan.css' WHERE `gl_name` = 'css_header'
 ALTER TABLE `users` DROP COLUMN `ssi_relayhealth`;
 #EndIf
 
+#IfNotRow4D supported_external_dataloads load_type ICD10 load_source CMS load_release_date 2016-10-01 load_filename 2017-PCS-Long-Abbrev-Titles.zip
+INSERT INTO upported_external_dataloads (oad_type, oad_source, oad_release_date, oad_filename, oad_checksum) VALUES ('ICD10', 'CMS', '2016-10-01', '2017-PCS-Long-Abbrev-Titles.zip', '4669c47f6a9ca34bf4c14d7f93b37993');
```
Comment (bradymiller) [inline sql/4_2_2-to-4_3_1_upgrade.sql]: Missing some leading l's :)

File: sql/database.sql [hunk 2]
```diff
@@ -6928,12 +6936,12 @@ CREATE TABLE code_types (
   PRIMARY KEY (ct_key)
 ) ENGINE=InnoDB;
 
-INSERT INTO code_types (ct_key, ct_id, ct_seq, ct_mod, ct_just, ct_fee, ct_rel, ct_nofs, ct_diag, ct_active, ct_label, ct_external, ct_claim, ct_proc, ct_term, ct_problem ) VALUES ('ICD9' , 2, 1, 0, ''    , 0, 0, 0, 1, 1, 'ICD9 Diagnosis', 4, 1, 0, 0, 1);
+INSERT INTO code_types (ct_key, ct_id, ct_seq, ct_mod, ct_just, ct_fee, ct_rel, ct_nofs, ct_diag, ct_active, ct_label, ct_external, ct_claim, ct_proc, ct_term, ct_problem ) VALUES ('ICD9' , 102, 102, 0, ''    , 0, 0, 0, 1, 1, 'ICD9 Diagnosis', 4, 1, 0, 0, 1);
```
Comment (bradymiller) [inline sql/database.sql]: Why this change?

File: sql/database.sql [hunk 3]
```diff
@@ -6928,12 +6936,12 @@ CREATE TABLE code_types (
...
 INSERT INTO code_types (ct_key, ct_id, ct_seq, ct_mod, ct_just, ct_fee, ct_rel, ct_nofs, ct_diag, ct_active, ct_label, ct_external, ct_claim, ct_proc, ct_term, ct_problem ) VALUES ('CVX'  , 100, 100, 0, '', 0, 0, 1, 0, 1, 'CVX Immunization', 0, 0, 0, 0, 0);
 INSERT INTO code_types (ct_key, ct_id, ct_seq, ct_mod, ct_just, ct_fee, ct_rel, ct_nofs, ct_diag, ct_active, ct_label, ct_external, ct_claim, ct_proc, ct_term, ct_problem ) VALUES ('DSMIV' , 101, 101, 0, '', 0, 0, 0, 1, 0, 'DSMIV Diagnosis', 0, 1, 0, 0, 1);
-INSERT INTO code_types (ct_key, ct_id, ct_seq, ct_mod, ct_just, ct_fee, ct_rel, ct_nofs, ct_diag, ct_active, ct_label, ct_external, ct_claim, ct_proc, ct_term, ct_problem ) VALUES ('ICD10' , 102, 102, 0, '', 0, 0, 0, 1, 1, 'ICD10 Diagnosis', 1, 1, 0, 0, 1);
+INSERT INTO code_types (ct_key, ct_id, ct_seq, ct_mod, ct_just, ct_fee, ct_rel, ct_nofs, ct_diag, ct_active, ct_label, ct_external, ct_claim, ct_proc, ct_term, ct_problem ) VALUES ('ICD10' , 2, 1, 0, '', 0, 0, 0, 1, 1, 'ICD10 Diagnosis', 1, 1, 0, 0, 1);
```
Comment (bradymiller) [inline sql/database.sql]: Why this change?

File: library/standard_tables_capture.inc [hunk 4]
```diff
@@ -327,7 +327,7 @@ function icd_import($type) {
     if( is_dir($dir) && $handle = opendir($dir)) {
         while (false !== ($filename = readdir($handle))) {
 	    // bypass unwanted entries
-	    if (!stripos($filename, ".txt") || stripos($filename,"diff")) {
+	    if (!stripos($filename, ".txt") || stripos($filename,"diff") || stripos($filename, "addenda")) {
```
Comment (bradymiller) [inline library/standard_tables_capture.inc]: What's in the addenda?

File: sql/database.sql (hunk 2 above)
Comment (stephenwaite) [inline sql/database.sql]: we can probably make icd10 the first entry in the dropdown search for diag code types

File: sql/4_2_2-to-4_3_1_upgrade.sql (hunk 1 above)
Comment (stephenwaite) [inline sql/4_2_2-to-4_3_1_upgrade.sql]: search and replace fail :(

File: library/standard_tables_capture.inc (hunk 4 above)
Comment (stephenwaite) [inline library/standard_tables_capture.inc]: here's a snippet, i don't think we need to program for this but i'll test in an upgrade to see if it makes a difference

Add:         1 C49A9   Gastrointestinal stromal tumor of other sites                Gastrointestinal stromal tumor of other sites
Revise from: 1 C7A094  Malignant carcinoid tumor of the foregut NOS                 Malignant carcinoid tumor of the foregut NOS
Revise to:   1 C7A094  Malignant carcinoid tumor of the foregut, unspecified        Malignant carcinoid tumor of the foregut, unspecified

Comment (stephenwaite) [inline library/standard_tables_capture.inc]: i upgraded from the 2016 set to 2017 by manualing adding the files to the contrib directory and issuing the raw sql update inside phpmyadmin, as you can see from the attached file, the addenda are not importing in the proper manner and are upsetting the selection of active=1 in the table too

[_icd10_dx_order_code___phpMyAdmin_4_5_4.pdf](https://github.com/openemr/openemr/files/431051/_icd10_dx_order_code___phpMyAdmin_4_5_4.pdf)
